apps/shell/app.js

```diff
@@ -55,9 +56,67 @@ app.use(process.env.PASSENGER_BASE_URI || '/', router);
 var server = new http.createServer(app);
 var wss = new WebSocket.Server({ noServer: true });
 
+
+
+whitelist = process.env.SSHHOST_WHITELIST;
+if (whitelist === undefined){
+  whitelist = [];
+} else {
+  whitelist = whitelist.split(':');
+}
+
+var filter = '.yml'; // add regex capability
+var files = fs.readdirSync('/etc/ood/config/clusters.d');
+files.forEach( function (file){
+  if (file.indexOf(filter) >= 0){
+    var fullPath = '/etc/ood/config/clusters.d/' + file;
+    var data = yaml.safeLoad(fs.readFileSync(fullPath)); //data is in json format
+    var errorMessage = 'Cluster config file does not follow correct v2 schema. Please see osc.github.io/ood-documentation.';
+
+    if (data.hasOwnProperty('v2')){
+      var v2 = data.v2;
```

Comment:
@matthu017 https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment Destructuring assignments can be useful here too.

Here's an example:
```js
const someObject = {
  foo: 'bar',
  hello: {
    world: true
  }
}

const { someKey = 'someDefault', hello: { world } } = someObject

console.log(world) // true
console.log(someKey) // 'someDefault'
```